clover.rb

```diff
@@ -3,38 +3,688 @@
+    otp_disable_notice_flash "One-time password authentication has been disabled"
+    otp_disable_error_flash "Error disabling one-time password authentication"
+    otp_disable_redirect { "/" + two_factor_manage_route }
+
+    # OTP Auth
+    otp_auth_view { view "auth/otp_auth", "One-Time" }
+    otp_auth_button "Authenticate Using One-Time Password"
+    otp_auth_link_text "One-Time Password Generator"
+
+    # Webauthn Setup
+    webauthn_setup_route "account/multifactor/webauthn-setup"
+    webauthn_setup_view { view "account/multifactor/webauthn_setup", "My Account" }
+    webauthn_setup_link_text "Add"
+    webauthn_setup_button "Setup Security Key"
+    webauthn_setup_notice_flash "Security key is now setup, please make note of your recovery codes"
+    webauthn_setup_error_flash "Error setting up security key"
+    webauthn_key_insert_hash { |credential| super(credential).merge(name: request.params["name"]) }
+
+    # :nocov:
+    after_webauthn_setup do
+      flash["notice"] = webauthn_setup_notice_flash
+      redirect "/" + recovery_codes_route
+    end
+    # :nocov:
+
+    # Webauthn Remove
+    webauthn_remove_route "account/multifactor/webauthn-remove"
+    webauthn_remove_view { view "account/multifactor/webauthn_remove", "My Account" }
+    webauthn_remove_link_text "Remove"
+    webauthn_remove_button "Remove Security Key"
+    webauthn_remove_notice_flash "Security key has been removed"
+    webauthn_remove_error_flash "Error removing security key"
+    webauthn_invalid_remove_param_message "Invalid security key to remove"
+    webauthn_remove_redirect { "/" + two_factor_manage_route }
+
+    # Webauthn Auth
+    webauthn_auth_view { view "auth/webauthn_auth", "Security Keys" }
+    webauthn_auth_button "Authenticate Using Security Keys"
+    webauthn_auth_link_text "Security Keys"
+
+    # Recovery Codes
+    recovery_codes_route "account/multifactor/recovery-codes"
+    recovery_codes_view { view "account/multifactor/recovery_codes", "My Account" }
+    recovery_codes_link_text "View"
+    add_recovery_codes_view { view "account/multifactor/recovery_codes", "My Account" }
+    auto_add_recovery_codes? true
+    auto_remove_recovery_codes? true
+    recovery_auth_view { view "auth/recovery_auth", "Recovery Codes" }
+  end
+
+  hash_branch("dashboard") do |r|
+    view "/dashboard"
+  end
+
+  hash_branch("after-login") do |r|
+    if (project = current_account.projects_dataset.order(:created_at, :name).first)
+      r.redirect "#{project.path}/dashboard"
+    else
+      r.redirect "/project"
+    end
+  end
+
+  # :nocov:
+  if Config.test?
+    hash_branch(:webhook_prefix, "test-error") do |r|
+      raise
+    end
+  end
+  # :nocov:
+
+  autoload_routes("web")
+  autoload_routes("api", "api")
+  autoload_routes("runtime", "runtime")
+
+  route do |r|
+    if api?
+      response.json = true
+      # To make test and development easier
+      # :nocov:
+      unless Config.production?
+        r.on("api") do
```

Comment:
@fdr https://github.com/ubicloud/ubicloud/pull/2084 will remove some branches from this PR

Reply:
After I merge this, I'll see if I can update #2084 for this and other changes.